models/surveys.go

```diff
@@ -486,15 +493,46 @@ func (api *API) Info(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+// AllSurveys returns a list of all known surveys
 func (api *API) AllSurveys(w http.ResponseWriter, r *http.Request) {
-	rows, err := api.AllSurveysStmt.Query()
+	var rows *sql.Rows
+	var err error
+	rows, err = api.AllSurveysStmt.Query()
+	if err != nil {
+		logError("Get all surveys returned error", err)
+		http.Error(w, "Failed to retrieve surveys", http.StatusInternalServerError)
+		return
+	}
+	parseSurveys(rows, w)
+}
 
+// SurveysByType returns surveys of a particular type
+func (api *API) SurveysByType(w http.ResponseWriter, r *http.Request) {
+	var rows *sql.Rows
+	var err error
+
+	urlSplit := strings.Split(r.URL.String(), "/")
+	surveyType := urlSplit[len(urlSplit)-1]
+
+	// Valid types are enumerated in DB
+	if surveyType != "Business" && surveyType != "Social" && surveyType != "Census" {
```

Comment:
Params should be probably case tolerant (Business == BUSINESS = business). Could also then just internally map and use that map to check validity
e.g:
```go
var surveyTypes = map[string]string{
	"business": "Business",
	"social": "Social,"
}

...

vars := mux.Vars(r)
t =: strings.ToLower(vars["type"])
if mapped, ok := surveyTypes[t]; ok {
  // Do query ..
  return		
}
// Error - unknown type
```